core/src/main/scala/org/apache/spark/scheduler/ResultTask.scala

```diff
@@ -65,4 +65,6 @@ private[spark] class ResultTask[T, U](
   override def preferredLocations: Seq[TaskLocation] = preferredLocs
 
   override def toString = "ResultTask(" + stageId + ", " + partitionId + ")"
+
+  override def canEqual(other: Any): Boolean = other.isInstanceOf[ResultTask[T, U]]
```

Comment:
You can try `Seq[Int](1).isInstanceOf[Seq[String]]` in REPL, it will return true.
`isInstanceOf` can't work on generic type because of JVM type erasure.